File: src/test/java/com/example/kehtolaulu/nanoblogs/test/AuthenticatedTest.java
package com.example.kehtolaulu.nanoblogs.test;

import com.example.kehtolaulu.nanoblogs.test.auth.AuthenticationPage;
import com.example.kehtolaulu.nanoblogs.test.auth.UserData;
import org.junit.After;
import org.junit.Before;
import org.junit.BeforeClass;

public abstract class AuthenticatedTest {
    private static AuthenticationPage authPage;
    private UserData userData;

    @BeforeClass
    public static void init() {
        if (authPage == null) {
            authPage = new AuthenticationPage();
        }
    }

    @Before
    public void logIn() {
        init();
        userData = new UserData("aivvent@yandex.ru", "123456");
        authPage.visit();
        authPage.login(userData);
    }

    public UserData getUserData() {
        return userData;
    }

    @After
    public void logOut() {
        authPage.logOut();
    }
}

File: src/test/java/com/example/kehtolaulu/nanoblogs/test/AuthenticatedTest.kt
package com.example.kehtolaulu.nanoblogs.test

import com.example.kehtolaulu.nanoblogs.CommonAccount
import com.example.kehtolaulu.nanoblogs.test.auth.AuthenticationPage
import com.example.kehtolaulu.nanoblogs.test.auth.UserData
import org.junit.Before
import org.junit.BeforeClass


abstract class AuthenticatedTest {
    val userData: UserData = CommonAccount

    @Before
    fun logIn() {
        init()
        if (!loggedIn()) {
            authPage.visit()
            authPage.login(userData)
        }
    }

    fun logOut() {
        authPage.logOut()
    }

    fun loggedIn() = authPage.hasText("sign out")

    companion object {
        private lateinit var authPage: AuthenticationPage
        @BeforeClass
        @JvmStatic
        fun init() {
            authPage = AuthenticationPage()
        }
    }
}
